<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.0.25">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Daniel Martinez-Martinez">

<title>E. coli pangenome metabolic capabilities</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="Main_report_files/libs/clipboard/clipboard.min.js"></script>
<script src="Main_report_files/libs/quarto-html/quarto.js"></script>
<script src="Main_report_files/libs/quarto-html/popper.min.js"></script>
<script src="Main_report_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Main_report_files/libs/quarto-html/anchor.min.js"></script>
<link href="Main_report_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Main_report_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Main_report_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Main_report_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Main_report_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

<link href="Main_report_files/libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="Main_report_files/libs/pagedtable-1.1/js/pagedtable.js"></script>


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">E. coli pangenome metabolic capabilities</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Daniel Martinez-Martinez </p>
          </div>
  </div>
    
    
  </div>
  

</header>

<section id="quarto" class="level2">
<h2 class="anchored" data-anchor-id="quarto">Quarto</h2>
</section>
<section id="background" class="level2">
<h2 class="anchored" data-anchor-id="background">Background</h2>
<p>We are animals walking in a world of microbes. Since the first moment we set foot on this planet, it had been inhabited by a myriad of other beings that had already mastered the cycles of Earth. Our interdependence with microbes is not only evolutionary but also ecological and functional. They affect our daily life in more ways that we could have ever imagined, in health and disease. But then the opposite is also true. We, the hosts, also affect them and can control their populations at some extent. This endless dance between the two parts is a hard topic to study due to the massive number of elements at a play. To add an extra layer of complexity, microbes are really diverse, and we can find several hundreds of different species in the human gut.</p>
<p>In a recent paper by Almeida <em>et al.</em><span class="citation" data-cites="Almeida2020"><sup><a href="#ref-Almeida2020" role="doc-biblioref">1</a></sup></span>, they cataloged a potential number of species that populate our guts, with around 4500 species characterized. More importantly is the fact that these species are divided in more than 200K different genomes, which points to the presence of more than one reference genome per species. This has profound implications in this context as within the same species we can find a huge genetic diversity that must be explored if we want to study microbe-host interactions in a natural context. The genetic diversity found within a species defines what is known by a <strong>pangenome</strong>, which can be described as the collection of different organisms within a single taxonomic level. The concept has been extensively used to describe the genetic diversity of species, describing how diverse it is. In this regard, we can distinguish between two ends of the spectrum: the open pangenomes, where the accessory genome (i.e., the genes that are strain-specific) is very large in comparison with the core genome (i.e., the genes that are shared by almost every strain); and the closed pangenome where the size of the accessory genome is much smaller. Although we can find some examples in-between, this remains a good way of classifying pangenomes.</p>
<p>This framework has been used to study the <em>E. coli</em> pangenome and determining that it is an open pangenome<span class="citation" data-cites="Touchon2020"><sup><a href="#ref-Touchon2020" role="doc-biblioref">2</a></sup></span> . This species has been under study for decades and it is still a source of unknown features. The <em>E. coli</em> species can be further divided in different phylogenetic groups, named as phylogroups. There are more than 10 phylogroups currently defined for <em>E. coli</em>, although usually we can find between 8 or 9 (A, B1, B2, C, D, E, F, G are the most common)<span class="citation" data-cites="Abram2021"><sup><a href="#ref-Abram2021" role="doc-biblioref">3</a></sup></span>. These groups have an evolutionary history that make them related also in their ecology, as for example the B2 group is usually enriched with pathogens. What this also defines is that their genomes and accessory genomes will be similar <strong>within</strong> phylogroups compared to <strong>between</strong> phylogroups.</p>
<p>[To follow with more info]</p>
</section>
<section id="data-exploration" class="level2">
<h2 class="anchored" data-anchor-id="data-exploration">Data exploration</h2>
<div class="cell" data-hash="Main_report_cache/html/unnamed-chunk-2_94527fa23215db18448a0bfdae822597">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># main data for the pathways</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>genome_paths <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="fu">here</span>(<span class="st">'tables'</span>,</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>                               <span class="st">'genome_pathways.xlsx'</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="Main_report_cache/html/data-reading_28b27d5c0a123824539f5667efc515c9">

</div>
<p>From the metabolic models generated by <code>gapseq</code><span class="citation" data-cites="Zimmermann2021"><sup><a href="#ref-Zimmermann2021" role="doc-biblioref">4</a></sup></span>, we can gather all the info into a big table that has this look:</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(genome_paths, <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["ID"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Name"],"name":[2],"type":["chr"],"align":["left"]},{"label":["Prediction"],"name":[3],"type":["lgl"],"align":["right"]},{"label":["Completeness"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["VagueReactions"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["KeyReactions"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["KeyReactionsFound"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["ReactionsFound"],"name":[8],"type":["chr"],"align":["left"]},{"label":["Genome"],"name":[9],"type":["chr"],"align":["left"]}],"data":[{"1":"|12DICHLORETHDEG-PWY|","2":"1,2-dichloroethane degradation","3":"FALSE","4":"25","5":"0","6":"1","7":"0","8":"ALDXANAU-RXN","9":"1"},{"1":"|14DICHLORBENZDEG-PWY|","2":"1,4-dichlorobenzene degradation","3":"FALSE","4":"28","5":"0","6":"2","7":"0","8":"DIENELACHYDRO-RXN MALEYLACETATE-REDUCTASE-RXN","9":"1"},{"1":"|1CMET2-PWY|","2":"folate transformations III (E. coli)","3":"TRUE","4":"100","5":"0","6":"0","7":"0","8":"DIHYDROFOLATEREDUCT-RXN 5-FORMYL-THF-CYCLO-LIGASE-RXN 1.5.1.20-RXN HOMOCYSMETB12-RXN GCVMULTI-RXN METHENYLTHFCYCLOHYDRO-RXN METHYLENETHFDEHYDROG-NADP-RXN GLYOHMETRANS-RXN THYMIDYLATESYN-RXN","9":"1"},{"1":"|2AMINOBENZDEG-PWY|","2":"anthranilate degradation III (anaerobic)","3":"FALSE","4":"0","5":"0","6":"0","7":"0","8":"NA","9":"1"},{"1":"|2ASDEG-PWY|","2":"orthanilate degradation","3":"FALSE","4":"25","5":"1","6":"0","7":"0","8":"NA","9":"1"},{"1":"|2OXOBUTYRATECAT-PWY|","2":"2-oxobutanoate degradation II","3":"FALSE","4":"50","5":"0","6":"0","7":"0","8":"RXN-7938","9":"1"},{"1":"|2PHENDEG-PWY|","2":"phenylethylamine degradation I","3":"FALSE","4":"50","5":"0","6":"1","7":"0","8":"PHENDEHYD-RXN","9":"1"},{"1":"|3-HYDROXYPHENYLACETATE-DEGRADATION-PWY|","2":"4-hydroxyphenylacetate degradation","3":"FALSE","4":"42","5":"0","6":"1","7":"0","8":"CHMS-DEHYDROGENASE-RXN 4-HYDROXY-2-KETOPIMELATE-LYSIS-RXN SUCCSEMIALDDEHYDROG-RXN","9":"1"},{"1":"|4-HYDROXYMANDELATE-DEGRADATION-PWY|","2":"4-hydroxymandelate degradation","3":"FALSE","4":"50","5":"0","6":"1","7":"1","8":"4-HYDROXYMANDELATE-OXY-RXN HYDROXYBENZALDEHYDE-OXIDATION-NAD-RXN HYDROXYBENZALDEHYDE-OXIDATION-NADP-RXN","9":"1"},{"1":"|4TOLCARBDEG-PWY|","2":"4-toluenecarboxylate degradation","3":"FALSE","4":"33","5":"0","6":"0","7":"0","8":"CYDTER-RXN","9":"1"},{"1":"|6-HYDROXYCINEOLE-DEGRADATION-PWY|","2":"1,8-cineole degradation","3":"FALSE","4":"0","5":"3","6":"0","7":"0","8":"NA","9":"1"},{"1":"|ACETOACETATE-DEG-PWY|","2":"acetoacetate degradation (to acetyl CoA)","3":"TRUE","4":"100","5":"0","6":"1","7":"1","8":"ACETYL-COA-ACETYLTRANSFER-RXN ACETOACETYL-COA-TRANSFER-RXN","9":"1"},{"1":"|ADENOSYLHOMOCYSCAT-PWY|","2":"L-methionine salvage from L-homocysteine","3":"FALSE","4":"66","5":"0","6":"0","7":"0","8":"2.1.1.5-RXN HOMOCYSMETB12-RXN","9":"1"},{"1":"|AEROBACTINSYN-PWY|","2":"aerobactin biosynthesis","3":"FALSE","4":"0","5":"0","6":"0","7":"0","8":"NA","9":"1"},{"1":"|ALACAT2-PWY|","2":"L-alanine degradation II (to D-lactate)","3":"TRUE","4":"100","5":"0","6":"0","7":"0","8":"ALANINE-AMINOTRANSFERASE-RXN DLACTDEHYDROGNAD-RXN GLUTAMATE-DEHYDROGENASE-RXN","9":"1"},{"1":"|ALADEG-PWY|","2":"L-alanine degradation I","3":"TRUE","4":"100","5":"0","6":"1","7":"1","8":"ALARACECAT-RXN DALADEHYDROG-RXN","9":"1"},{"1":"|ALANINE-DEG3-PWY|","2":"L-alanine degradation III","3":"TRUE","4":"100","5":"0","6":"0","7":"0","8":"ALANINE-AMINOTRANSFERASE-RXN","9":"1"},{"1":"|ALANINE-SYN2-PWY|","2":"L-alanine biosynthesis II","3":"TRUE","4":"100","5":"0","6":"0","7":"0","8":"ALANINE-AMINOTRANSFERASE-RXN","9":"1"},{"1":"|ALANINE-VALINESYN-PWY|","2":"L-alanine biosynthesis I","3":"TRUE","4":"100","5":"0","6":"0","7":"0","8":"BRANCHED-CHAINAMINOTRANSFERVAL-RXN VALINE-PYRUVATE-AMINOTRANSFER-RXN","9":"1"},{"1":"|ALKANEMONOX-PWY|","2":"two-component alkanesulfonate monooxygenase","3":"TRUE","4":"100","5":"0","6":"1","7":"1","8":"RXN-12444 RXN0-280","9":"1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>The different columns from the table give us information about the status of each pathway by genome or strain. We can then see the distribution of how many complete pathways do we have per strain, as seen in <a href="#fig-comp-paths-histo">Figure&nbsp;1</a> . By the look of this histogram, seems that we are dealing with a bimodal distribution.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>genome_paths <span class="sc">%&gt;%</span> </span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Completeness <span class="sc">==</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Name, Genome) <span class="sc">%&gt;%</span> </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Genome) <span class="sc">%&gt;%</span> </span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">&gt;</span> <span class="dv">400</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(n)) <span class="sc">+</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">color =</span> <span class="st">'black'</span>, <span class="at">fill =</span> <span class="st">'darkblue'</span>, <span class="at">bins =</span> <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'Number of complete pathways in genome'</span>,</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'Number of genomes'</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-comp-paths-histo" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="Main_report_files/figure-html/fig-comp-paths-histo-1.png" class="img-fluid figure-img" width="480"></p>
<p></p><figcaption class="figure-caption">Figure 1: Histogram of complete pathways per genome in our cohort</figcaption><p></p>
</figure>
</div>
</div>
</div>
<!-- -->


</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body" data-line-spacing="2" role="doc-bibliography">
<div id="ref-Almeida2020" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Almeida, A. <em>et al.</em> <a href="https://doi.org/10.1038/s41587-020-0603-3">A unified catalog of 204, 938 reference genomes from the human gut microbiome</a>. <em>Nature Biotechnology</em> <strong>39</strong>, 105–114 (2020).</div>
</div>
<div id="ref-Touchon2020" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Touchon, M. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pgen.1008866">Phylogenetic background and habitat drive the genetic diversification of escherichia coli</a>. <em><span>PLOS</span> Genetics</em> <strong>16</strong>, e1008866 (2020).</div>
</div>
<div id="ref-Abram2021" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Abram, K. <em>et al.</em> <a href="https://doi.org/10.1038/s42003-020-01626-5">Mash-based analyses of escherichia coli genomes reveal 14 distinct phylogroups</a>. <em>Communications Biology</em> <strong>4</strong>, (2021).</div>
</div>
<div id="ref-Zimmermann2021" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Zimmermann, J., Kaleta, C. &amp; Waschina, S. <a href="https://doi.org/10.1186/s13059-021-02295-1">Gapseq: Informed prediction of bacterial metabolic pathways and reconstruction of accurate metabolic models</a>. <em>Genome Biology</em> <strong>22</strong>, (2021).</div>
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb4" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "E. coli pangenome metabolic capabilities"</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="an">bibliography:</span><span class="co"> references.bib</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="an">csl:</span><span class="co"> nature.csl</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Daniel Martinez-Martinez"</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span><span class="co"> </span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="co">  pdf: </span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co">    echo: false</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a><span class="fu">## Quarto</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-packages</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(circlize)</span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggsankey)</span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vcd)</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue)</span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_cowplot</span>(<span class="dv">15</span>))</span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a><span class="fu">## Background</span></span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a>We are animals walking in a world of microbes. Since the first moment we set foot on this planet, it had been inhabited by a myriad of other beings that had already mastered the cycles of Earth. Our interdependence with microbes is not only evolutionary but also ecological and functional. They affect our daily life in more ways that we could have ever imagined, in health and disease. But then the opposite is also true. We, the hosts, also affect them and can control their populations at some extent. This endless dance between the two parts is a hard topic to study due to the massive number of elements at a play. To add an extra layer of complexity, microbes are really diverse, and we can find several hundreds of different species in the human gut.</span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a>In a recent paper by Almeida *et al.* @Almeida2020, they cataloged a potential number of species that populate our guts, with around 4500 species characterized. More importantly is the fact that these species are divided in more than 200K different genomes, which points to the presence of more than one reference genome per species. This has profound implications in this context as within the same species we can find a huge genetic diversity that must be explored if we want to study microbe-host interactions in a natural context. The genetic diversity found within a species defines what is known by a **pangenome**, which can be described as the collection of different organisms within a single taxonomic level. The concept has been extensively used to describe the genetic diversity of species, describing how diverse it is. In this regard, we can distinguish between two ends of the spectrum: the open pangenomes, where the accessory genome (i.e., the genes that are strain-specific) is very large in comparison with the core genome (i.e., the genes that are shared by almost every strain); and the closed pangenome where the size of the accessory genome is much smaller. Although we can find some examples in-between, this remains a good way of classifying pangenomes.</span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a>This framework has been used to study the *E. coli* pangenome and determining that it is an open pangenome @Touchon2020 . This species has been under study for decades and it is still a source of unknown features. The *E. coli* species can be further divided in different phylogenetic groups, named as phylogroups. There are more than 10 phylogroups currently defined for *E. coli*, although usually we can find between 8 or 9 (A, B1, B2, C, D, E, F, G are the most common) @Abram2021. These groups have an evolutionary history that make them related also in their ecology, as for example the B2 group is usually enriched with pathogens. What this also defines is that their genomes and accessory genomes will be similar **within** phylogroups compared to **between** phylogroups.</span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a><span class="sc">\[</span>To follow with more info<span class="sc">\]</span></span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a><span class="fu">## Data exploration</span></span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a><span class="co">#| cache: true</span></span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a><span class="co"># main data for the pathways</span></span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a>genome_paths <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="fu">here</span>(<span class="st">'tables'</span>,</span>
<span id="cb4-63"><a href="#cb4-63" aria-hidden="true" tabindex="-1"></a>                               <span class="st">'genome_pathways.xlsx'</span>))</span>
<span id="cb4-64"><a href="#cb4-64" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-65"><a href="#cb4-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-68"><a href="#cb4-68" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-69"><a href="#cb4-69" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb4-70"><a href="#cb4-70" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: data-reading</span></span>
<span id="cb4-71"><a href="#cb4-71" aria-hidden="true" tabindex="-1"></a><span class="co">#| cache: true</span></span>
<span id="cb4-72"><a href="#cb4-72" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb4-73"><a href="#cb4-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-74"><a href="#cb4-74" aria-hidden="true" tabindex="-1"></a><span class="co"># get the genome names</span></span>
<span id="cb4-75"><a href="#cb4-75" aria-hidden="true" tabindex="-1"></a>gnm_names <span class="ot">=</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="st">"./raw_data/pathways"</span>, <span class="at">pattern =</span> <span class="st">".tbl"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-76"><a href="#cb4-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_sub</span>(<span class="at">start =</span> <span class="dv">1</span>, <span class="at">end =</span> <span class="sc">-</span><span class="dv">18</span>)</span>
<span id="cb4-77"><a href="#cb4-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-78"><a href="#cb4-78" aria-hidden="true" tabindex="-1"></a><span class="co"># get the files names</span></span>
<span id="cb4-79"><a href="#cb4-79" aria-hidden="true" tabindex="-1"></a>files_list <span class="ot">=</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="st">"./raw_data/pathways"</span>, <span class="at">pattern =</span> <span class="st">".tbl"</span>)</span>
<span id="cb4-80"><a href="#cb4-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-81"><a href="#cb4-81" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb4-82"><a href="#cb4-82" aria-hidden="true" tabindex="-1"></a><span class="co"># # main data for the pathways</span></span>
<span id="cb4-83"><a href="#cb4-83" aria-hidden="true" tabindex="-1"></a><span class="co"># genome_paths = read_excel(here('tables',</span></span>
<span id="cb4-84"><a href="#cb4-84" aria-hidden="true" tabindex="-1"></a><span class="co">#                                'genome_pathways.xlsx'))</span></span>
<span id="cb4-85"><a href="#cb4-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-86"><a href="#cb4-86" aria-hidden="true" tabindex="-1"></a><span class="co"># metadata from the genomes</span></span>
<span id="cb4-87"><a href="#cb4-87" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">"~/Documents/MRC_postdoc/Pangenomic/metadata/MAIN_metadata.xlsx"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-88"><a href="#cb4-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Genome =</span> <span class="fu">str_sub</span>(fasta, <span class="at">end =</span> <span class="sc">-</span><span class="dv">7</span>), <span class="at">.before =</span> Strainname) <span class="sc">%&gt;%</span> </span>
<span id="cb4-89"><a href="#cb4-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Genome <span class="sc">%in%</span> gnm_names) <span class="sc">%&gt;%</span> </span>
<span id="cb4-90"><a href="#cb4-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(phylogroup <span class="sc">!=</span> <span class="st">'cladeI'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-91"><a href="#cb4-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Genome, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-92"><a href="#cb4-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-93"><a href="#cb4-93" aria-hidden="true" tabindex="-1"></a><span class="co"># pathways-reactions metadata </span><span class="al">###</span></span>
<span id="cb4-94"><a href="#cb4-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-95"><a href="#cb4-95" aria-hidden="true" tabindex="-1"></a>metacyc_paths <span class="ot">=</span> <span class="fu">read_delim</span>(<span class="st">"../tables/All-reactions-of-MetaCyc_extended.txt"</span>, </span>
<span id="cb4-96"><a href="#cb4-96" aria-hidden="true" tabindex="-1"></a>                           <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, </span>
<span id="cb4-97"><a href="#cb4-97" aria-hidden="true" tabindex="-1"></a>                           <span class="at">trim_ws =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-98"><a href="#cb4-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">ec_number =</span> <span class="st">`</span><span class="at">EC-Number</span><span class="st">`</span>,</span>
<span id="cb4-99"><a href="#cb4-99" aria-hidden="true" tabindex="-1"></a>         <span class="at">ID =</span> <span class="st">`</span><span class="at">Object ID</span><span class="st">`</span>,</span>
<span id="cb4-100"><a href="#cb4-100" aria-hidden="true" tabindex="-1"></a>         <span class="at">reaction =</span> Reaction,</span>
<span id="cb4-101"><a href="#cb4-101" aria-hidden="true" tabindex="-1"></a>         <span class="at">substrates =</span> <span class="st">`</span><span class="at">Substrates</span><span class="st">`</span>,</span>
<span id="cb4-102"><a href="#cb4-102" aria-hidden="true" tabindex="-1"></a>         <span class="at">common_name =</span> <span class="st">`</span><span class="at">Common-Name</span><span class="st">`</span>,</span>
<span id="cb4-103"><a href="#cb4-103" aria-hidden="true" tabindex="-1"></a>         <span class="at">names =</span> Names,</span>
<span id="cb4-104"><a href="#cb4-104" aria-hidden="true" tabindex="-1"></a>         <span class="at">spontaneous =</span> <span class="st">`</span><span class="at">Spontaneous?</span><span class="st">`</span>,</span>
<span id="cb4-105"><a href="#cb4-105" aria-hidden="true" tabindex="-1"></a>         <span class="at">systematic_name=</span> <span class="st">`</span><span class="at">Systematic-Name</span><span class="st">`</span>,</span>
<span id="cb4-106"><a href="#cb4-106" aria-hidden="true" tabindex="-1"></a>         <span class="at">pathway =</span> <span class="st">`</span><span class="at">In-Pathway</span><span class="st">`</span>)</span>
<span id="cb4-107"><a href="#cb4-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-108"><a href="#cb4-108" aria-hidden="true" tabindex="-1"></a><span class="co"># ModelSeed comps </span><span class="al">###</span></span>
<span id="cb4-109"><a href="#cb4-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-110"><a href="#cb4-110" aria-hidden="true" tabindex="-1"></a>modelSeed_reactions <span class="ot">=</span> <span class="fu">read_delim</span>(<span class="st">"https://raw.githubusercontent.com/ModelSEED/ModelSEEDDatabase/master/Biochemistry/reactions.tsv"</span>, </span>
<span id="cb4-111"><a href="#cb4-111" aria-hidden="true" tabindex="-1"></a>           <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, </span>
<span id="cb4-112"><a href="#cb4-112" aria-hidden="true" tabindex="-1"></a>           <span class="at">trim_ws =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-113"><a href="#cb4-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-114"><a href="#cb4-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-115"><a href="#cb4-115" aria-hidden="true" tabindex="-1"></a>modelSeed_comps <span class="ot">=</span> <span class="fu">read_delim</span>(<span class="st">"https://raw.githubusercontent.com/ModelSEED/ModelSEEDDatabase/master/Biochemistry/compounds.tsv"</span>, </span>
<span id="cb4-116"><a href="#cb4-116" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, </span>
<span id="cb4-117"><a href="#cb4-117" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">trim_ws =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-118"><a href="#cb4-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-119"><a href="#cb4-119" aria-hidden="true" tabindex="-1"></a><span class="co"># filter metadata for the bugs we want to analyse: AUS and ECOREF</span></span>
<span id="cb4-120"><a href="#cb4-120" aria-hidden="true" tabindex="-1"></a>meta_filt <span class="ot">=</span> metadata <span class="sc">%&gt;%</span> </span>
<span id="cb4-121"><a href="#cb4-121" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Genome =</span> <span class="fu">str_sub</span>(fasta, <span class="at">start =</span> <span class="dv">1</span>, <span class="at">end =</span> <span class="sc">-</span><span class="dv">7</span>), <span class="at">.before =</span> ID) <span class="sc">%&gt;%</span> </span>
<span id="cb4-122"><a href="#cb4-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Discard <span class="sc">==</span> <span class="st">'No'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-123"><a href="#cb4-123" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Origin <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">'AUS'</span>, <span class="st">'ECOREF'</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-124"><a href="#cb4-124" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Genome, <span class="at">.keep_all =</span> T) </span>
<span id="cb4-125"><a href="#cb4-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-126"><a href="#cb4-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-127"><a href="#cb4-127" aria-hidden="true" tabindex="-1"></a><span class="co"># collection of reactions from the pangenome</span></span>
<span id="cb4-128"><a href="#cb4-128" aria-hidden="true" tabindex="-1"></a>pan_reactions <span class="ot">=</span> genome_paths <span class="sc">%&gt;%</span> </span>
<span id="cb4-129"><a href="#cb4-129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(Completeness)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-130"><a href="#cb4-130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(ID, <span class="at">.keep_all =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb4-131"><a href="#cb4-131" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(ReactionsFound, <span class="at">sep =</span> <span class="st">' '</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-132"><a href="#cb4-132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(ReactionsFound) <span class="sc">%&gt;%</span> </span>
<span id="cb4-133"><a href="#cb4-133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(ReactionsFound)</span>
<span id="cb4-134"><a href="#cb4-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-135"><a href="#cb4-135" aria-hidden="true" tabindex="-1"></a>metacyc_paths_pangenome <span class="ot">=</span> metacyc_paths <span class="sc">%&gt;%</span> </span>
<span id="cb4-136"><a href="#cb4-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(reaction <span class="sc">%in%</span> pan_reactions)  <span class="sc">%&gt;%</span> </span>
<span id="cb4-137"><a href="#cb4-137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(substrates, </span>
<span id="cb4-138"><a href="#cb4-138" aria-hidden="true" tabindex="-1"></a>                <span class="at">sep =</span> <span class="st">' // '</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-139"><a href="#cb4-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">met_name =</span> substrates)</span>
<span id="cb4-140"><a href="#cb4-140" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-141"><a href="#cb4-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-142"><a href="#cb4-142" aria-hidden="true" tabindex="-1"></a>From the metabolic models generated by <span class="in">`gapseq`</span>@Zimmermann2021, we can gather all the info into a big table that has this look:</span>
<span id="cb4-143"><a href="#cb4-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-146"><a href="#cb4-146" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-147"><a href="#cb4-147" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(genome_paths, <span class="dv">20</span>)</span>
<span id="cb4-148"><a href="#cb4-148" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-149"><a href="#cb4-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-150"><a href="#cb4-150" aria-hidden="true" tabindex="-1"></a>The different columns from the table give us information about the status of each pathway by genome or strain. We can then see the distribution of how many complete pathways do we have per strain, as seen in @fig-comp-paths-histo . By the look of this histogram, seems that we are dealing with a bimodal distribution.</span>
<span id="cb4-151"><a href="#cb4-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-154"><a href="#cb4-154" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-155"><a href="#cb4-155" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-comp-paths-histo</span></span>
<span id="cb4-156"><a href="#cb4-156" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Histogram of complete pathways per genome in our cohort</span></span>
<span id="cb4-157"><a href="#cb4-157" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 5</span></span>
<span id="cb4-158"><a href="#cb4-158" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb4-159"><a href="#cb4-159" aria-hidden="true" tabindex="-1"></a>genome_paths <span class="sc">%&gt;%</span> </span>
<span id="cb4-160"><a href="#cb4-160" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Completeness <span class="sc">==</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-161"><a href="#cb4-161" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Name, Genome) <span class="sc">%&gt;%</span> </span>
<span id="cb4-162"><a href="#cb4-162" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Genome) <span class="sc">%&gt;%</span> </span>
<span id="cb4-163"><a href="#cb4-163" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb4-164"><a href="#cb4-164" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">&gt;</span> <span class="dv">400</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-165"><a href="#cb4-165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(n)) <span class="sc">+</span></span>
<span id="cb4-166"><a href="#cb4-166" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">color =</span> <span class="st">'black'</span>, <span class="at">fill =</span> <span class="st">'darkblue'</span>, <span class="at">bins =</span> <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb4-167"><a href="#cb4-167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb4-168"><a href="#cb4-168" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'Number of complete pathways in genome'</span>,</span>
<span id="cb4-169"><a href="#cb4-169" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'Number of genomes'</span></span>
<span id="cb4-170"><a href="#cb4-170" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb4-171"><a href="#cb4-171" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>